cvmfs/authz/authz_curl.cc

```diff
@@ -98,6 +98,36 @@ CURLcode AuthzAttachment::CallbackSslCtx(
 }
 
 
+bool AuthzAttachment::ConfigureSciTokenCurl(
```

Comment:
I think it would be better to create another class `AuthzOauthAttachement` derived from `download::CredentialsAttachment`.  The current `AuthzAttachement` should then probably renamed to `AuthzX509Attachement`.  Some slight adjustment between base class and derived classes is probably necessary in order to line things up in the `MountPoint` class.

Reply:
I looked into the `MountPoint` class.  At that time, when it creates the `AuthzAttachement`, it doesn't know what type of authentication it is going to be returned by the helper.  Also, there are circumstances, such as when we are transitioning LIGO from x509 to Bearer tokens, when the authentication plugin may be able to return both.

Reply:
Hm, that's right.  Thanks for checking.  Please ignore the comment.